content/blog/learn-tidyverse-ai/index.Rmd

```diff
@@ -0,0 +1,542 @@
+
+And here is the result:
+
+> Here's how to accomplish the same tasks using tidyverse functions:
+
+```{r}
+#| eval: false
+clubs <- clubs %>%
+  mutate(membership_status = if_else(is.na(membership_status), 
+                                   "Closed", 
+                                   "Open")) %>%
+  relocate(membership_status)
+
+clubs %>% 
+  head(10)
+```
+
+> This tidyverse solution:
+>
+> -   Uses mutate() with `if_else()` to recode the variable
+> -   Uses relocate() to move membership_status to the front
+> -   Shows the first 10 rows using head()
+>
+> The tidyverse version is more readable and follows the principle of chaining operations with the pipe operator (`%>%`).
+> `if_else()` is the tidyverse equivalent of base R's `ifelse()`, with stricter type checking.
+>
+> Would you like me to explain any of these functions in more detail?
+
+We're getting there!
+For new learners of the tidyverse, there are still a few aspects of this code that might be misleading:
+
+-   It doesn't use the base pipe operator (`|>`), which is the pipe operator used in R for Data Science.
+-   Line breaks and indentation does not follow [tidyverse style guidelines](https://style.tidyverse.org/pipes.html#long-lines), which suggest that if the arguments to a function don't all fit on one line, each argument should go on its own line.
+-   If `clubs` is a tibble, printing it will only display the first 10 rows anyway, so there's no need for the `head()` call.
+
+Once again, it's possible to ask the tool to make these updates.
+Claude doesn't provide a sharing link for the conversation, but I've saved it at <https://gist.github.com/mine-cetinkaya-rundel/c74dc7c5f2a27a846574edd063988431>.
+Even with vague prompts like "If clubs is a tibble, it'll only print out the first 10 rows anyway," it does exactly what we would want it to do, resulting in the following:
+
+```{r}
+#| eval: false
+clubs <- clubs |>
+  mutate(membership_status = if_else(is.na(membership_status), "Closed", "Open")) |>
+  relocate(membership_status)
+
+clubs
+```
+
+The operative word here being "want".
+Once again, someone with enough experience with R and the tidyverse would not explicitly ask for these improvements; they would just make them themselves and carry on.
+But someone who is just learning the tidyverse would likely stop much earlier.
+In my experience, many new learners stop after the very first prompt, which looks unfamiliar, and hence potentially intimidating.
+For learners of the tidyverse, the simple "Can you do this with the tidyverse?" can be a very useful prompt improvement that gets them majority of the way there.
+
+## Case study 3: Web scraping
+
+For the third and last example we'll try scraping some data from the web, specifically some [structured North Carolina weather data](https://www.usclimatedata.com/climate/durham/north-carolina/united-states/usnc0192). 
+For this example, I've used [perplexity.ai](https://www.perplexity.ai/).
```

Comment:
Maybe mentioning why perplexity? (Presumably because it will actually download the web page?)